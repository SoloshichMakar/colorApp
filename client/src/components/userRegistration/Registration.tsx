import React from "react";
import { NavLink, Redirect } from "react-router-dom";

interface IRegistrationProps {
  email: string;
  password: string;
  confirmPassword: string;
  message: string;
  isCreated: boolean;
  userIsCreate:Function;
  error: Function;
  textEmailChange: Function;
  textPasswordChange: Function;
  textPasswordConfirmChange: Function;
  createUser: Function;
}

const Registration: React.FC<IRegistrationProps> = ({
  email,
  password,
  confirmPassword,
  message,
  createUser,
  isCreated,
  error,
  textEmailChange,
  textPasswordChange,
  textPasswordConfirmChange,
  userIsCreate,
}) => {
  function currentTextEmailChange(e: React.ChangeEvent<HTMLInputElement>) {
    const getTextAreaValue = e.target.value;
    textEmailChange(getTextAreaValue);
  }

  function currentTextPasswordChange(e: React.ChangeEvent<HTMLInputElement>) {
    const getTextAreaValue = e.target.value;
    textPasswordChange(getTextAreaValue);
  }

  function currentTextConfirmPasswordChange(
    e: React.ChangeEvent<HTMLInputElement>
  ) {
    const getTextAreaValue = e.target.value;
    textPasswordConfirmChange(getTextAreaValue);
  }

  function createNewUser() {
    if(password.length > 5) {
      if (password === confirmPassword) {
        createUser(email, password);
      } else {
        error("Password and confirm password does not match");
      }
    } else {
      error("Password must be at least six characters long");
    }
  }

  if (isCreated) {
    userIsCreate();
    error();
    return <Redirect to="/login" />;
  }

  return (
    <div className="registration_main">
      <div>Registration Page</div>
      <div className="registration_main__message_box">
        <div className="registration_main__message_label" id="message">
          {message}
        </div>
        <div className="registration_main__email_box">
          <div className="registration_main__email_label">Email</div>
          <input
            className="registration_main__email_input"
            type="text"
            id="email"
            name="email"
            value={email}
            onChange={currentTextEmailChange}
          />
        </div>
        <div className="registration_main__password_box">
          <div className="registration_main__password_label">Password</div>
          <input
            className="registration_main__password_input"
            type="password"
            id="password"
            value={password}
            onChange={currentTextPasswordChange}
          />
          <div className="registration_main__confirm_password_box">
            <div className="registration_main__confirm_password_label">
              Confirm Password
            </div>
            <input
              className="registration_main__confirm_password_input"
              type="password"
              id="confirmPassword"
              value={confirmPassword}
              onChange={currentTextConfirmPasswordChange}
            />
          </div>
        </div>
      </div>
      <input
        className="registration_main__create_user_button"
        type="button"
        value="addUser"
        onClick={createNewUser}
      />
      <NavLink className="registration_main__back_link" to="/login">
        Back
      </NavLink>
    </div>
  );
};

export default Registration;
