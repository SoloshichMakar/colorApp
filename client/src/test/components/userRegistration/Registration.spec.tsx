import React from "react";
import renderer from "react-test-renderer";

import Registration from "../../../components/userRegistration/Registration";
import { BrowserRouter } from "react-router-dom";

const props = {
  message: "Test Messsage",
  email: "Test@mail.com",
  password: "Test Password",
  confirmPassword: "Test Password",
  error: Function,
  textEmailChange: Function,
  textPasswordChange: Function,
  textPasswordConfirmChange: Function,
  createUser: Function,
};

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Registration
          message={props.message}
          email={props.email}
          password={props.password}
          confirmPassword={props.confirmPassword}
          textEmailChange={props.textEmailChange}
          textPasswordChange={props.textPasswordChange}
          textPasswordConfirmChange={props.textPasswordConfirmChange}
          createUser={props.createUser}
          error={props.error}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
