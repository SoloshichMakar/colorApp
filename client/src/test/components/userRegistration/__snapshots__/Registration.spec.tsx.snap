// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="registration_main"
>
  <div>
    Registration Page
  </div>
  <div
    className="registration_main__message_box"
  >
    <div
      className="registration_main__message_label"
      id="message"
    >
      Test Messsage
    </div>
    <div
      className="registration_main__email_box"
    >
      <div
        className="registration_main__email_label"
      >
        Email
      </div>
      <input
        className="registration_main__email_input"
        id="email"
        name="email"
        onChange={[Function]}
        type="text"
        value="Test@mail.com"
      />
    </div>
    <div
      className="registration_main__password_box"
    >
      <div
        className="registration_main__password_label"
      >
        Password
      </div>
      <input
        className="registration_main__password_input"
        id="password"
        onChange={[Function]}
        type="password"
        value="Test Password"
      />
      <div
        className="registration_main__confirm_password_box"
      >
        <div
          className="registration_main__confirm_password_label"
        >
          Confirm Password
        </div>
        <input
          className="registration_main__confirm_password_input"
          id="confirmPassword"
          onChange={[Function]}
          type="password"
          value="Test Password"
        />
      </div>
    </div>
  </div>
  <input
    className="registration_main__create_user_button"
    onClick={[Function]}
    type="button"
    value="addUser"
  />
  <a
    aria-current={null}
    className="registration_main__back_link"
    href="/login"
    onClick={[Function]}
  >
    Back
  </a>
</div>
`;
